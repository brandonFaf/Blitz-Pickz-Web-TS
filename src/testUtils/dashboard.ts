export const Dashboard0 = {
  rankings: [
    {
      points: 0,
      rank: 1
    }
  ]
};
export const Dashboard1 = {
  rankings: [
    {
      points: 8,
      rank: 2
    }
  ]
};
export const Dashboard7 = {
  rankings: [
    {
      points: 38,
      rank: 2
    }
  ]
};
export const Dashboard17 = {
  rankings: [
    {
      points: 140,
      rank: 1
    }
  ]
};
