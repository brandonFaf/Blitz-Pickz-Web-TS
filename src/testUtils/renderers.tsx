//@ts-nocheck
import React from 'react';
import { MockedProvider } from '@apollo/client/testing';
import { UserContext } from '../contexts/UserContext';
import { GroupContext } from '../contexts/GroupContext';
import { Router } from 'react-router-dom';
import { createMemoryHistory } from 'history';
export const RenderApollo = ({
  children,
  mock = [],
  group = {},
  user = {}
}) => {
  const history = createMemoryHistory();
  return (
    <MockedProvider mocks={mock} addTypename={false}>
      <UserContext.Provider value={{ user }}>
        <GroupContext.Provider value={{ group }}>
          <Router history={history}>{children}</Router>
        </GroupContext.Provider>
      </UserContext.Provider>
    </MockedProvider>
  );
};
