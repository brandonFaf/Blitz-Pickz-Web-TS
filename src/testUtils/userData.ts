export default [
  {
    display_name: 'Allison Bell',
    id: '1',
    photo_url: 'https://randomuser.me/api/portraits/thumb/women/14.jpg'
  },
  {
    display_name: 'Michael Johnson',
    id: 'd76e9371-0ee5-4be1-8e28-291c53950c2c',
    photo_url: 'https://randomuser.me/api/portraits/thumb/men/37.jpg'
  },
  {
    display_name: 'Vera Lynch',
    id: 'dbaf7003-2cc5-4982-bef0-c537c5cff40a',
    photo_url: 'https://randomuser.me/api/portraits/thumb/women/50.jpg'
  },
  {
    display_name: 'Fred Ramos',
    id: 'af84f088-4dcd-46cc-a128-3179b8a6d522',
    photo_url: 'https://randomuser.me/api/portraits/thumb/men/50.jpg'
  },
  {
    display_name: 'Gabriel Rice',
    id: '30748f78-1964-48be-be26-4b325b657a5d',
    photo_url: 'https://randomuser.me/api/portraits/thumb/men/24.jpg'
  },
  {
    display_name: 'Fletcher Brown',
    id: '01722379-1afa-4835-a5e8-19355cc5a7ef',
    photo_url: 'https://randomuser.me/api/portraits/thumb/men/12.jpg'
  },
  {
    display_name: 'Urbano da Mota',
    id: '5efbe1d6-c686-434d-bfcf-f58350837cd0',
    photo_url: 'https://randomuser.me/api/portraits/thumb/men/46.jpg'
  },
  {
    display_name: 'Lonnie Martin',
    id: '83fd618d-c0a6-4264-90c2-86261fe426ac',
    photo_url: 'https://randomuser.me/api/portraits/thumb/men/3.jpg'
  },
  {
    display_name: 'Bertha Phillips',
    id: 'b1e47119-1e94-49a3-a368-6b5577fa74ea',
    photo_url: 'https://randomuser.me/api/portraits/thumb/women/70.jpg'
  },
  {
    display_name: 'Sílvia Oliveira',
    id: '95c46f04-e55e-4d3a-87e8-7fbd30848d26',
    photo_url: 'https://randomuser.me/api/portraits/thumb/women/52.jpg'
  },
  {
    display_name: 'Mandy Holland',
    id: '37507393-feaf-4000-9b96-3a6c749f8f2f',
    photo_url: 'https://randomuser.me/api/portraits/thumb/women/52.jpg'
  },
  {
    display_name: 'Ron Richardson',
    id: 'b0048d10-fef3-41c8-ad8f-200b0878326b',
    photo_url: 'https://randomuser.me/api/portraits/thumb/men/65.jpg'
  },
  {
    display_name: 'Berilo Castro',
    id: '24096123-7384-4057-ba8e-b4bf29e5adad',
    photo_url: 'https://randomuser.me/api/portraits/thumb/men/55.jpg'
  },
  {
    display_name: 'Larry Wagner',
    id: '33f4d2a6-82b1-46ed-aead-79f0ca49d458',
    photo_url: 'https://randomuser.me/api/portraits/thumb/men/16.jpg'
  },
  {
    display_name: 'Teófilo Moreira',
    id: 'd14037df-c34d-4dd7-a06b-8a890fb28655',
    photo_url: 'https://randomuser.me/api/portraits/thumb/men/85.jpg'
  },
  {
    display_name: 'Maia Edwards',
    id: 'c36c3ccb-7d7d-4c1a-bd5d-8ca284cfebaa',
    photo_url: 'https://randomuser.me/api/portraits/thumb/women/42.jpg'
  },
  {
    display_name: 'William Lewis',
    id: '20f52c88-b2d0-4d20-a179-4d8c6136c06b',
    photo_url: 'https://randomuser.me/api/portraits/thumb/men/76.jpg'
  },
  {
    display_name: 'Francis Caldwell',
    id: 'b753b6d5-6119-4e20-a08a-012cc444d257',
    photo_url: 'https://randomuser.me/api/portraits/thumb/men/33.jpg'
  },
  {
    display_name: 'Christy Graham',
    id: '2864914e-4429-41ad-8abb-0639cc4de6ea',
    photo_url: 'https://randomuser.me/api/portraits/thumb/women/39.jpg'
  },
  {
    display_name: 'Otacília Mendes',
    id: 'a65b5568-8546-44cf-9b60-d7dc2398c48b',
    photo_url: 'https://randomuser.me/api/portraits/thumb/women/83.jpg'
  },
  {
    display_name: 'Maia Green',
    id: 'a7afa0da-78fa-421a-b6eb-64059ce19435',
    photo_url: 'https://randomuser.me/api/portraits/thumb/women/1.jpg'
  },
  {
    display_name: 'Jake Jones',
    id: '57e49d5e-513f-4d5f-bcb3-b4485f784015',
    photo_url: 'https://randomuser.me/api/portraits/thumb/men/14.jpg'
  },
  {
    display_name: 'Martha Fox',
    id: '7ebbf438-ea45-42c2-ac97-9aac02a3591f',
    photo_url: 'https://randomuser.me/api/portraits/thumb/women/5.jpg'
  },
  {
    display_name: 'Claudia Ruiz',
    id: '12351915-c95b-468d-819a-9e453b5e0cdb',
    photo_url: 'https://randomuser.me/api/portraits/thumb/women/57.jpg'
  },
  {
    display_name: 'Jeremiah Edwards',
    id: '90f71d8b-31b8-4e88-9ee6-f4dfe35ae284',
    photo_url: 'https://randomuser.me/api/portraits/thumb/men/30.jpg'
  },
  {
    display_name: 'Joann Roberts',
    id: '39749110-365e-4ac6-a71b-1ef85ce63c9d',
    photo_url: 'https://randomuser.me/api/portraits/thumb/women/15.jpg'
  },
  {
    display_name: 'Marlene Ramos',
    id: 'ba3541c5-bec9-4457-a509-cf026c432db2',
    photo_url: 'https://randomuser.me/api/portraits/thumb/women/50.jpg'
  },
  {
    display_name: 'Pascoal Melo',
    id: 'dc47463d-d4c8-4987-adc6-8e6ea31c7ba1',
    photo_url: 'https://randomuser.me/api/portraits/thumb/men/11.jpg'
  },
  {
    display_name: 'Emily Grant',
    id: 'f9e4f801-a1b1-4d0d-b86d-5f857b9a529c',
    photo_url: 'https://randomuser.me/api/portraits/thumb/women/76.jpg'
  },
  {
    display_name: 'Eddie Wells',
    id: '58685a39-a1cc-40df-b723-d62e45221152',
    photo_url: 'https://randomuser.me/api/portraits/thumb/men/18.jpg'
  },
  {
    display_name: 'Avery Sims',
    id: '3ed3133c-25dc-49ee-a464-a9cfa882aa06',
    photo_url: 'https://randomuser.me/api/portraits/thumb/women/62.jpg'
  },
  {
    display_name: 'Wesley Hudson',
    id: 'c062d06a-c5bf-470a-8100-f4f35d4b1532',
    photo_url: 'https://randomuser.me/api/portraits/thumb/men/90.jpg'
  },
  {
    display_name: 'Elijah Bell',
    id: 'f3c4c877-ded5-44d5-a0a3-6071492256aa',
    photo_url: 'https://randomuser.me/api/portraits/thumb/men/32.jpg'
  },
  {
    display_name: 'Alma Gordon',
    id: '8889aa92-e9ca-4292-a465-673c541e2f25',
    photo_url: 'https://randomuser.me/api/portraits/thumb/women/5.jpg'
  },
  {
    display_name: 'Floyd Phillips',
    id: 'b51f6a26-c447-43a5-ba2d-87ccfd9c4daa',
    photo_url: 'https://randomuser.me/api/portraits/thumb/men/45.jpg'
  },
  {
    display_name: 'Caroline Lucas',
    id: 'd6cfbfbc-262d-4025-86d0-256ff3a3cca9',
    photo_url: 'https://randomuser.me/api/portraits/thumb/women/4.jpg'
  },
  {
    display_name: 'Susie Patterson',
    id: '31a5dcab-ef8b-456f-9fb0-a1f528fa1d85',
    photo_url: 'https://randomuser.me/api/portraits/thumb/women/5.jpg'
  },
  {
    display_name: 'Priscilla Larson',
    id: '0bc2bac2-b5a8-4c21-be7c-7b4695c99fad',
    photo_url: 'https://randomuser.me/api/portraits/thumb/women/54.jpg'
  },
  {
    display_name: 'Clarisa Vieira',
    id: 'f5118fbb-d2c9-4921-b373-ef80b4d2a05c',
    photo_url: 'https://randomuser.me/api/portraits/thumb/women/6.jpg'
  },
  {
    display_name: 'Aquila Moreira',
    id: 'ab3e4516-6346-4399-95e7-24ab1ce1ca72',
    photo_url: 'https://randomuser.me/api/portraits/thumb/men/75.jpg'
  },
  {
    display_name: 'Sofia Davies',
    id: 'ed4eef1a-d55f-495a-ad2a-6de4cc74f672',
    photo_url: 'https://randomuser.me/api/portraits/thumb/women/92.jpg'
  },
  {
    display_name: 'Ramona Howard',
    id: 'd94b8f4b-b467-4d22-a097-fbffd338ff5d',
    photo_url: 'https://randomuser.me/api/portraits/thumb/women/85.jpg'
  },
  {
    display_name: 'Wilma Dixon',
    id: '8b09dd51-9df1-4d57-8fb9-e54dacd02cba',
    photo_url: 'https://randomuser.me/api/portraits/thumb/women/55.jpg'
  },
  {
    display_name: 'Adrian Knight',
    id: 'da84e810-b325-4ceb-9d95-27236c91bd0f',
    photo_url: 'https://randomuser.me/api/portraits/thumb/men/99.jpg'
  },
  {
    display_name: 'Alice Flores',
    id: '6b6e600d-c7f9-46da-bc14-e39226bf4942',
    photo_url: 'https://randomuser.me/api/portraits/thumb/women/27.jpg'
  },
  {
    display_name: 'Noah Hughes',
    id: '21bd8f4b-6dcf-4bc2-808b-49b5ead539c7',
    photo_url: 'https://randomuser.me/api/portraits/thumb/men/32.jpg'
  },
  {
    display_name: 'Genésio Aragão',
    id: 'e02c5813-fbcd-4b80-be3f-e06681912fd0',
    photo_url: 'https://randomuser.me/api/portraits/thumb/men/75.jpg'
  },
  {
    display_name: 'Barry Gonzalez',
    id: '057b15b1-9aea-496b-96a9-403c0c6c9796',
    photo_url: 'https://randomuser.me/api/portraits/thumb/men/31.jpg'
  },
  {
    display_name: 'Felix Robertson',
    id: 'd073f811-c8eb-4301-a895-6145b3a4f9f0',
    photo_url: 'https://randomuser.me/api/portraits/thumb/men/47.jpg'
  },
  {
    display_name: 'Nathaniel Li',
    id: 'adf9ee90-a2f7-4da3-8ce7-14b8e048f79e',
    photo_url: 'https://randomuser.me/api/portraits/thumb/men/42.jpg'
  }
];
