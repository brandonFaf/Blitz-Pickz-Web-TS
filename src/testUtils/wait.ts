import { act } from 'react-dom/test-utils';

export default async () =>
  await act(async () => {
    await new Promise(resolve => setTimeout(resolve, 10));
  });
