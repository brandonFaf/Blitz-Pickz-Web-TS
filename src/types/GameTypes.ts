import { Picks, User } from './graphql.types';

export type PickModel = Pick<
  Picks,
  'id' | 'selected_id' | 'group_id' | 'correct'
> & {
  user: Pick<User, 'id' | 'display_name'>;
};
