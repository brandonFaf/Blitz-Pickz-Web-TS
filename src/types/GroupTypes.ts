import {
  Group,
  User,
  Group_User_Aggregate_Fields,
  Maybe,
  Rankings
} from './graphql.types';
export type MemberModel = Pick<User, 'photo_url' | 'id' | 'display_name'>;
export type GroupModel = Pick<Group, 'display_name' | 'id' | 'admin_id'> & {
  members: Array<{ user: MemberModel }>;
};
export type GroupDrawerModel = Pick<
  Group,
  'display_name' | 'id' | 'admin_id'
> & {
  rankings: Array<Pick<Rankings, 'rank' | 'points'>>;
  members: Array<{ user: Pick<User, 'photo_url' | 'id' | 'display_name'> }>;
};
export type SearchGroupModel = Pick<
  Group,
  'id' | 'admin_id' | 'display_name' | 'private' | 'passcode'
> & {
  members_aggregate: {
    aggregate?: Maybe<Pick<Group_User_Aggregate_Fields, 'count'>>;
  };
};
export enum JoinGroupStages {
  search,
  passcode,
  join
}
export type JoinGroupParams = {
  group: SearchGroupModel;
  navigate: (nextStage: JoinGroupStages, g: SearchGroupModel | null) => void;
};
