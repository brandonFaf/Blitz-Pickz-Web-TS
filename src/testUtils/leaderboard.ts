import userData from './userData';
export const Leaderboard1 = {
  group_by_pk: {
    id: 1,
    display_name: 'test group 1'
  },
  rankings: [
    {
      points: 40,
      rank: 1,
      user: userData[0]
    }
  ]
};

export const Leaderboard2 = {
  group_by_pk: {
    id: 1,
    display_name: 'test group 2'
  },
  rankings: [
    {
      points: 40,
      rank: 1,
      user: userData[0]
    },
    {
      points: 38,
      rank: 2,
      user: userData[1]
    }
  ]
};
export const Leaderboard3 = {
  group_by_pk: {
    id: 1,
    display_name: 'Avengers'
  },
  rankings: [
    {
      points: 40,
      rank: 1,
      user: userData[0]
    },
    {
      points: 38,
      rank: 2,
      user: userData[1]
    },
    {
      points: 37,
      rank: 3,
      user: userData[2]
    },
    {
      points: 37,
      rank: 3,
      user: userData[3]
    },
    {
      points: 32,
      rank: 5,
      user: userData[4]
    },
    {
      points: 31,
      rank: 6,
      user: userData[5]
    },
    {
      points: 27,
      rank: 7,
      user: userData[6]
    }
  ]
};
