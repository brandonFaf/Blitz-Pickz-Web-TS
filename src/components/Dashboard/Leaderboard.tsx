import React, { useEffect } from 'react';
import styled from 'styled-components/macro';
import ProfilePhoto from '../Profile/ProfilePhoto';
import Colors from '../../Styles/colors';
import useUser from '../../hooks/useUser';
import useGroup from '../../hooks/useGroup';
import { useLeaderboardLazyQuery } from '../../types/graphql.types';

const Row = styled.div`
  display: grid;
  grid-template-columns: 5vw 45px 28vw 10vw;
  grid-auto-flow: row;
  grid-column-gap: 12px;
  align-items: center;
  .current {
    color: ${Colors.highlight};
  }
`;

const LBoard = styled.div`
  display: grid;
  grid-auto-flow: row;
  margin-top: 40px;
  font-size: 14px;
  justify-content: center;
  text-align: center;
  align-items: center;
  grid-row-gap: 10px;
  overflow-y: scroll;
  margin: 0 10vw;
  grid-auto-rows: max-content;
`;

const Leaderboard = () => {
  const { user } = useUser();
  const { group } = useGroup();
  const [
    loadLeaderboard,
    { data, loading, error, called }
  ] = useLeaderboardLazyQuery();
  useEffect(() => {
    if (group) {
      loadLeaderboard({
        variables: { group_id: group.id }
      });
    }
  }, [group, loadLeaderboard]);
  if (!called || loading) {
    return <div>Loading...</div>;
  }
  if (error) {
    console.log(error);
    return <div>There was a problem loading this data</div>;
  }
  if (!data || !data.group_by_pk) {
    return <div></div>;
  }
  const members =
    data.rankings.length === 0
      ? group?.members.map(m => ({ ...m, rank: 1, points: 0 }))
      : data.rankings;
  return (
    <LBoard data-testid='leaderboard'>
      <Row>
        <div />
        <div />
        <div>Player</div>
        <div>Points</div>
      </Row>
      {members?.map(m => {
        if (!m.user) {
          return <></>;
        }
        const { photo_url, display_name, id } = m.user;
        let cn = '';
        if (id === user.id) {
          cn = 'current';
        }
        return (
          <Row data-testid='leaderboard-row' key={id} className={cn}>
            <div>{m.rank || 1}.</div>
            <div>
              <ProfilePhoto
                size={'small'}
                displayName={display_name}
                src={photo_url}
              />
            </div>
            <div className={cn}>{display_name}</div>
            <div>{m.points || 0}</div>
          </Row>
        );
      })}
    </LBoard>
  );
};

export default Leaderboard;
