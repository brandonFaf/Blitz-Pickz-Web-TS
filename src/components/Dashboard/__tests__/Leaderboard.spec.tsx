import React from 'react';
import { RenderApollo } from '../../../testUtils/renderers';
import Leaderboard from '../Leaderboard';
import { LeaderboardDocument } from '../../../types/graphql.types';
import { Leaderboard1, Leaderboard3 } from '../../../testUtils/leaderboard';
import userData from '../../../testUtils/userData';
import { render, cleanup } from '@testing-library/react';
import wait from '../../../testUtils/wait';

afterEach(cleanup);
const renderLeaderboard = (mock, group, user) => {
  const props = { mock, group, user };
  return render(
    <RenderApollo {...props}>
      <Leaderboard />
    </RenderApollo>
  );
};
describe('<Leaderboard/>', () => {
  it('renders correctly for one person in the list', async () => {
    //setup
    const group = { id: 1 };
    const user = { id: '1' };
    const mock = [
      {
        request: {
          query: LeaderboardDocument,
          variables: {
            group_id: group.id
          }
        },
        result: { data: Leaderboard1 }
      }
    ];
    const { asFragment } = renderLeaderboard(mock, group, user);
    await wait();
    expect(asFragment()).toMatchSnapshot();
  });
  it('renders correctly for many people in the list', async () => {
    const group = { id: 1 };
    const user = { id: '1' };
    const mock = [
      {
        request: {
          query: LeaderboardDocument,
          variables: {
            group_id: group.id
          }
        },
        result: { data: Leaderboard3 }
      }
    ];
    const { asFragment } = renderLeaderboard(mock, group, user);
    await wait();
    expect(asFragment()).toMatchSnapshot();
  });
  it('renders group members if data rankings is empty', async () => {
    const group = {
      id: 2,
      members: [{ user: userData[1] }, { user: userData[0] }]
    };
    const user = { id: '1' };
    const mock = [
      {
        request: {
          query: LeaderboardDocument,
          variables: {
            group_id: group.id
          }
        },
        result: { data: { ...Leaderboard1, rankings: [] } }
      }
    ];
    const { asFragment } = renderLeaderboard(mock, group, user);
    await wait();
    expect(asFragment()).toMatchSnapshot();
  });
});
