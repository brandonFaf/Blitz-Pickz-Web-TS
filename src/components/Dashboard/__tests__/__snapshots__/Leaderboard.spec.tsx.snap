// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Leaderboard/> renders correctly for many people in the list 1`] = `
<DocumentFragment>
  <div
    class="Leaderboard__LBoard-sc-15t7lnt-1 jeYLWJ"
    data-testid="leaderboard"
  >
    <div
      class="Leaderboard__Row-sc-15t7lnt-0 dOHGgg"
    >
      <div />
      <div />
      <div>
        Player
      </div>
      <div>
        Points
      </div>
    </div>
    <div
      class="Leaderboard__Row-sc-15t7lnt-0 dOHGgg current"
      data-testid="leaderboard-row"
    >
      <div>
        1.
      </div>
      <div>
        <div
          alt="profile"
          class="ProfilePhoto__PhotoImg-sc-1wpt521-0 dHqMbu"
          src="https://randomuser.me/api/portraits/thumb/women/14.jpg"
        />
      </div>
      <div
        class="current"
      >
        Allison Bell
      </div>
      <div>
        40
      </div>
    </div>
    <div
      class="Leaderboard__Row-sc-15t7lnt-0 dOHGgg"
      data-testid="leaderboard-row"
    >
      <div>
        2.
      </div>
      <div>
        <div
          alt="profile"
          class="ProfilePhoto__PhotoImg-sc-1wpt521-0 bBetzn"
          src="https://randomuser.me/api/portraits/thumb/men/37.jpg"
        />
      </div>
      <div
        class=""
      >
        Michael Johnson
      </div>
      <div>
        38
      </div>
    </div>
    <div
      class="Leaderboard__Row-sc-15t7lnt-0 dOHGgg"
      data-testid="leaderboard-row"
    >
      <div>
        3.
      </div>
      <div>
        <div
          alt="profile"
          class="ProfilePhoto__PhotoImg-sc-1wpt521-0 dpNfGy"
          src="https://randomuser.me/api/portraits/thumb/women/50.jpg"
        />
      </div>
      <div
        class=""
      >
        Vera Lynch
      </div>
      <div>
        37
      </div>
    </div>
    <div
      class="Leaderboard__Row-sc-15t7lnt-0 dOHGgg"
      data-testid="leaderboard-row"
    >
      <div>
        3.
      </div>
      <div>
        <div
          alt="profile"
          class="ProfilePhoto__PhotoImg-sc-1wpt521-0 hpevHm"
          src="https://randomuser.me/api/portraits/thumb/men/50.jpg"
        />
      </div>
      <div
        class=""
      >
        Fred Ramos
      </div>
      <div>
        37
      </div>
    </div>
    <div
      class="Leaderboard__Row-sc-15t7lnt-0 dOHGgg"
      data-testid="leaderboard-row"
    >
      <div>
        5.
      </div>
      <div>
        <div
          alt="profile"
          class="ProfilePhoto__PhotoImg-sc-1wpt521-0 crPlGd"
          src="https://randomuser.me/api/portraits/thumb/men/24.jpg"
        />
      </div>
      <div
        class=""
      >
        Gabriel Rice
      </div>
      <div>
        32
      </div>
    </div>
    <div
      class="Leaderboard__Row-sc-15t7lnt-0 dOHGgg"
      data-testid="leaderboard-row"
    >
      <div>
        6.
      </div>
      <div>
        <div
          alt="profile"
          class="ProfilePhoto__PhotoImg-sc-1wpt521-0 hwRSpk"
          src="https://randomuser.me/api/portraits/thumb/men/12.jpg"
        />
      </div>
      <div
        class=""
      >
        Fletcher Brown
      </div>
      <div>
        31
      </div>
    </div>
    <div
      class="Leaderboard__Row-sc-15t7lnt-0 dOHGgg"
      data-testid="leaderboard-row"
    >
      <div>
        7.
      </div>
      <div>
        <div
          alt="profile"
          class="ProfilePhoto__PhotoImg-sc-1wpt521-0 hZRxBl"
          src="https://randomuser.me/api/portraits/thumb/men/46.jpg"
        />
      </div>
      <div
        class=""
      >
        Urbano da Mota
      </div>
      <div>
        27
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<Leaderboard/> renders correctly for one person in the list 1`] = `
<DocumentFragment>
  <div
    class="Leaderboard__LBoard-sc-15t7lnt-1 jeYLWJ"
    data-testid="leaderboard"
  >
    <div
      class="Leaderboard__Row-sc-15t7lnt-0 dOHGgg"
    >
      <div />
      <div />
      <div>
        Player
      </div>
      <div>
        Points
      </div>
    </div>
    <div
      class="Leaderboard__Row-sc-15t7lnt-0 dOHGgg current"
      data-testid="leaderboard-row"
    >
      <div>
        1.
      </div>
      <div>
        <div
          alt="profile"
          class="ProfilePhoto__PhotoImg-sc-1wpt521-0 dHqMbu"
          src="https://randomuser.me/api/portraits/thumb/women/14.jpg"
        />
      </div>
      <div
        class="current"
      >
        Allison Bell
      </div>
      <div>
        40
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`<Leaderboard/> renders group members if data rankings is empty 1`] = `
<DocumentFragment>
  <div
    class="Leaderboard__LBoard-sc-15t7lnt-1 jeYLWJ"
    data-testid="leaderboard"
  >
    <div
      class="Leaderboard__Row-sc-15t7lnt-0 dOHGgg"
    >
      <div />
      <div />
      <div>
        Player
      </div>
      <div>
        Points
      </div>
    </div>
    <div
      class="Leaderboard__Row-sc-15t7lnt-0 dOHGgg"
      data-testid="leaderboard-row"
    >
      <div>
        1.
      </div>
      <div>
        <div
          alt="profile"
          class="ProfilePhoto__PhotoImg-sc-1wpt521-0 bBetzn"
          src="https://randomuser.me/api/portraits/thumb/men/37.jpg"
        />
      </div>
      <div
        class=""
      >
        Michael Johnson
      </div>
      <div>
        0
      </div>
    </div>
    <div
      class="Leaderboard__Row-sc-15t7lnt-0 dOHGgg current"
      data-testid="leaderboard-row"
    >
      <div>
        1.
      </div>
      <div>
        <div
          alt="profile"
          class="ProfilePhoto__PhotoImg-sc-1wpt521-0 dHqMbu"
          src="https://randomuser.me/api/portraits/thumb/women/14.jpg"
        />
      </div>
      <div
        class="current"
      >
        Allison Bell
      </div>
      <div>
        0
      </div>
    </div>
  </div>
</DocumentFragment>
`;
